docs/contracts.rst

```diff
@@ -1151,6 +1151,7 @@ Interfaces
 Interfaces are similar to abstract contracts, but they cannot have any functions implemented. There are further restrictions:
 
 - Cannot inherit other contracts or interfaces.
+- Cannot declare non-external functions.
```

Comment:
Perhaps we should rather phase it as `All declared functions have to be external` - and perhaps we should also list the "functions cannot be implemented" as part of this list - people might skip the paragraph above.

Reply:
👍  for the new wording.

Also since there are now so many differences, perhaps it makes no sense to elevate "one major difference" into the sentence and list the rest, so probably better to have a simple list.